### PR TITLE
[choreo] Add a way for users to package up diagnostic zips for support and overall logging improvement

### PR DESCRIPTION
**Pull request by oh-yes-0-fps**
Should be merged after #680

### CODE REVIEW

**Review by oh-yes-0-fps — COMMENTED** [review]
Should actually also wait on #601 as im going to cleanup error types a bit in this too

File: src-tauri/src/api.rs [hunk 1]
```diff
@@ -144,17 +171,65 @@ pub async fn generate_remote(
...
 #[tauri::command]
 pub fn cancel_all_remote_generators(app_handle: tauri::AppHandle) {
     let remote_resources = app_handle.state::<RemoteGenerationResources>();
     remote_resources.kill_all();
 }
+
+#[tauri::command]
+pub fn open_diagnostic_file(project: ProjectFile, trajs: Vec<TrajFile>) -> ChoreoResult<()> {
+    tracing::debug!("Opening diagnostic file");
+    let log_lines = if let Some(dir) = dirs::state_dir().map(|d| d.join("logs")) {
```
Comment (calcmogul) [inline src-tauri/src/api.rs:192]: The more standard place to put this is somewhere like state_dir()/choreo/log.

Comment (Gold856) [reply src-tauri/src/api.rs:192]: How does this work on Windows? As far as I know, state_dir() just returns None on Windows.

Comment (calcmogul) [reply src-tauri/src/api.rs:192]: It doesn't. We should fix that.

Comment (Gold856) [reply src-tauri/src/api.rs:192]: That seems like a good place to put it.

Comment (oh-yes-0-fps) [reply src-tauri/src/api.rs]: I feel like the diagnostic zips should go somewhere under temp.

File: src-tauri/src/api.rs [hunk 2]
```diff
@@ -141,20 +169,38 @@ pub async fn generate_remote(
...
 #[tauri::command]
 pub fn cancel_all_remote_generators(app_handle: tauri::AppHandle) {
     let remote_resources = app_handle.state::<RemoteGenerationResources>();
     remote_resources.kill_all();
 }
+
+#[tauri::command]
+pub fn open_diagnostic_file(project: ProjectFile, trajs: Vec<TrajFile>) -> TauriResult<()> {
+    tracing::debug!("Opening diagnostic file");
+    let log_lines = get_log_lines(dirs::data_local_dir().map(|d| d.join("logs")));
```
Comment (calcmogul) [inline src-tauri/src/api.rs:193]:
```suggestion
    let log_lines = get_log_lines(dirs::data_local_dir().map(|d| d.join("choreo/log")));
```

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Review by shueja — APPROVED** [review]

**Review by calcmogul — APPROVED** [review]

**Pull request merged**